src/deserialize/mod.rs

```diff
@@ -177,6 +186,128 @@ where
     }
 }
 
+impl<T> Deserialize for HashMap<String, T>
+where
+    T: Deserialize,
+{
+    fn deserialize(edn: &Edn) -> Result<Self, Error> {
+        match edn {
+            Edn::Map(_) => Ok(edn
+                .map_iter()
+                .ok_or_else(|| Error::Iter(format!("Could not create iter from {:?}", edn)))?
+                .map(|(key, e)| (key, Deserialize::deserialize(e).unwrap()))
+                .fold(HashMap::new(), |mut acc, (key, e)| {
+                    acc.insert(key.to_string(), e);
+                    acc
+                })),
+            Edn::NamespacedMap(ns, _) => Ok(edn
+                .map_iter()
+                .ok_or_else(|| Error::Iter(format!("Could not create iter from {:?}", edn)))?
+                .map(|(key, e)| {
+                    (
+                        if ns.starts_with(":") {
+                            ns.to_string() + "/" + key
+                        } else {
+                            String::from(":") + ns + "/" + key
+                        },
+                        Deserialize::deserialize(e).unwrap(),
+                    )
+                })
+                .fold(HashMap::new(), |mut acc, (key, e)| {
+                    acc.insert(key.to_string(), e);
+                    acc
+                })),
+            _ => Err(build_deserialize_error(
+                &edn,
+                std::any::type_name::<HashMap<String, T>>(),
+            )),
+        }
+    }
+}
+
+impl<T> Deserialize for BTreeMap<String, T>
+where
+    T: Deserialize,
+{
+    fn deserialize(edn: &Edn) -> Result<Self, Error> {
+        match edn {
+            Edn::Map(_) => Ok(edn
+                .map_iter()
+                .ok_or_else(|| Error::Iter(format!("Could not create iter from {:?}", edn)))?
+                .map(|(key, e)| (key, Deserialize::deserialize(e).unwrap()))
```

Comment:
Again, `from_edn` > `Deserialize::deserialize` and also unsafety of using `unwrap`.